aiozk/test/test_lock.py

```diff
@@ -0,0 +1,50 @@
+import asyncio
+import logging
+import pytest
+import types
+
+from aiozk.exc import TimeoutError
+from aiozk.states import States
+
+log = logging.getLogger(__name__)
+
+
+@pytest.mark.asyncio
+async def test_creation_failure_deadlock(zk, path):
+    lock = zk.recipes.Lock(path)
+    await lock.ensure_path()
+
+    async def change_state():
+        await asyncio.sleep(1)
+        zk.session.state.transition_to(States.SUSPENDED)
+
+    async def create(self, *args, **kwargs):
+        await self.create_orig(*args, **kwargs)
+        # SHOULD NOT REACH HERE
+
+    zk.create_orig = zk.create
+    zk.create = types.MethodType(create, zk)
+
+    zk.session.conn.read_loop_task.cancel()
+    zk.session.conn.read_loop_task = None
+    # wait for that read loop task is cancelled
+    await asyncio.sleep(1)
+
+    asyncio.create_task(change_state())
+    lock_acquired = False
+    with pytest.raises(TimeoutError):
+        async with await lock.acquire(timeout=2):
+            lock_acquired = True
+
+    assert not lock_acquired
+    assert not lock.owned_paths
+
+    zk.create = zk.create_orig
+
+    lock2 = zk.recipes.Lock(path)
+    async with await lock2.acquire(timeout=2):
+        lock_acquired = True
+
+    assert lock_acquired
+
+    await zk.delete(path)
```

Comment:
This gives us quite noisy output with lot of errors, probably we should rewrite it to handle non-deleted path case. Maybe something like:
```python
    try:
        async with await lock2.acquire(timeout=2):
            lock_acquired = True
        assert lock_acquired
    finally:
        await zk.deleteall(path)
```

Reply:
Do you mean that lots of errors occur if fix is not applied?

Reply:
Yep, this makes the reading of error message painful and not so obvious

Reply:
After I changed the code using deleteall, error messages look better.